src/pq/param.cr

```diff
@@ -1,164 +1,292 @@
+require "../pg/interval"
 require "../pg/geo"
 
 module PQ
   # :nodoc:
-  record Param, slice : Slice(UInt8), size : Int32, format : Int16 do
+  record Param, slice : Slice(UInt8), size : Int32, format : Format do
+    enum Format : Int16
+      None   = -1
+      Text   =  0
+      Binary =  1
+    end
     delegate to_unsafe, to: slice
 
     #  Internal wrapper to represent an encoded parameter
 
     def self.encode(val : Nil)
-      binary Pointer(UInt8).null.to_slice(0), -1
+      binary Bytes.empty, -1
+    end
+
+    def self.encode(val : Bool, into slice : Bytes = Bytes.new(1))
```

Comment:
Being able to encode a value into an existing slice is important for arrays and also allowed me to encode things like intervals and geo types in terms of numeric types.